react/src/reactTextMask.jsx

```diff
@@ -1,64 +1,46 @@
 import React, {PropTypes} from 'react'
 import {
-  processComponentChanges,
+  processComponentChanges as processComponentChangesHelper,
   safeSetSelection,
   getComponentInitialState
-} from '../../core/src/componentHelpers.js'
+} from '../../core/src/reactComponentHelpers.js'
 
 export const MaskedInput = React.createClass({
   propTypes: {
     mask: PropTypes.string.isRequired,
     guide: PropTypes.bool,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+    value: PropTypes.string
   },
 
-  getInitialState() {
-    const {
-      value: inputValue,
-      mask,
-      guide,
-      placeholderCharacter: placeholderChar,
-      validator
-    } = this.props
+  componentWillMount() {
+    const {props: {mask, value, placeholderCharacter: placeholderChar}} = this
 
-    return getComponentInitialState({inputValue, mask, validator, guide, placeholderChar})
-  },
-
-  componentWillReceiveProps(nextProps) {
-    if (
-      nextProps.mask !== this.props.mask ||
-      nextProps.guide !== this.props.guide ||
-      nextProps.placeholderCharacter !== this.props.placeholderCharacter ||
-      nextProps.value !== undefined && nextProps.value !== this.state.conformedInput ||
-      this.props.value !== undefined && nextProps.value === undefined ||
-      nextProps.validator !== this.props.validator
-    ) {
-      const {
-        mask,
-        value: inputValue,
-        guide,
-        placeholderCharacter: placeholderChar,
-        validator
-      } = nextProps
-
-      this.setState(getComponentInitialState({mask, validator, inputValue, guide, placeholderChar}))
-    }
+    this.textMaskState = getComponentInitialState({mask, placeholderChar, value})
   },
 
   componentDidUpdate() {
-    safeSetSelection(this.inputElement, this.state.adjustedCaretPosition)
+    safeSetSelection(this.inputElement, this.textMaskState.adjustedCaretPosition)
   },
 
   render() {
-    const {props, state: {componentPlaceholder, conformedInput}, onChange} = this
-    const {placeholder = componentPlaceholder, type = 'text'} = props
+    const {
+      props, onChange, processComponentChanges,
+      textMaskState: {conformedInput, componentPlaceholder},
+      props: {value = conformedInput, type = 'text', placeholder = componentPlaceholder},
+    } = this
+
+    const finalConformedInput = (value !== conformedInput) ?
+      processComponentChanges(value, true).conformedInput :
+      value
+
+    this.textMaskState.conformedInput = finalConformedInput
```

Comment:
@browniefed I needed to store state outside of `setState` in order to be able to do this ^

That way the component stores the `conformedInput` and uses it as `previousConformedInput` in the next call.